bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreCatalog.java

```diff
@@ -0,0 +1,366 @@
+    // TODO: Enable once supported by BigQuery API.
+    throw new UnsupportedOperationException("Table rename operation is unsupported.");
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    Dataset builder = new Dataset();
+    DatasetReference datasetReference = toDatasetReference(namespace);
+    builder.setLocation(this.projectLocation);
+    builder.setDatasetReference(datasetReference);
+    builder.setExternalCatalogDatasetOptions(
+        BigQueryMetastoreUtils.createExternalCatalogDatasetOptions(
+            createDefaultStorageLocationUri(datasetReference.getDatasetId()), metadata));
+
+    client.create(builder);
+  }
+
+  @Override
+  public List<Namespace> listNamespaces() {
+    try {
+      return listNamespaces(Namespace.empty());
+    } catch (NoSuchNamespaceException e) {
+      return ImmutableList.of();
+    }
+  }
+
+  /**
+   * Since this catalog only supports one-level namespaces, it always returns an empty list unless
+   * passed an empty namespace to list all namespaces within the catalog.
+   */
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    List<Datasets> allDatasets = client.list(projectId);
+
+    ImmutableList<Namespace> namespaces =
+        allDatasets.stream().map(this::toNamespace).collect(ImmutableList.toImmutableList());
+    if (namespaces.isEmpty()) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    return namespaces;
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    try {
+      client.delete(toDatasetReference(namespace));
+      // We don't delete the data folder for safety, which aligns with Hive Metastore's default
+      // behavior.
+      // We can support database or catalog level config controlling file deletion in the future.
+      return true;
+    } catch (NoSuchNamespaceException e) {
+      return false;
+    }
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) {
+    Dataset dataset = client.load(toDatasetReference(namespace));
+
+    ExternalCatalogDatasetOptions existingOptions = dataset.getExternalCatalogDatasetOptions();
+    Map<String, String> existingParameters =
+        existingOptions != null ? existingOptions.getParameters() : null;
+
+    Map<String, String> newParameters = Maps.newHashMap();
+    if (existingParameters != null) {
+      newParameters.putAll(existingParameters);
+    }
+
+    newParameters.putAll(properties);
+
+    if (Objects.equals(existingParameters, newParameters)) {
+      // No change in parameters detected
+      return false;
+    }
+
+    client.setParameters(toDatasetReference(namespace), properties);
+    return true;
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) {
+    client.removeParameters(toDatasetReference(namespace), properties);
+    return true;
```

Comment:
This isn't really adhering to the API behavior because it's returning true if the properties are actually removed.  It appears that HiveCatalog has this same behavior, but JDBCCatalog does it correctly.

Reply:
Done The code follows JDBC's behaviour now. because i did not touch this line. The comment does not shown as outdated.